syntax = "proto3";

package streaming;

message FrameMessage {
  string serial_number = 1;
  int64 timestamp = 2;
  int32 frame_id = 3;
  bytes image = 4;
  int32 roi_x = 5;
  int32 roi_y = 6;
  int32 roi_w = 7;
  int32 roi_h = 8;
}

service FrameStreamer {
  rpc SendFrame (FrameMessage) returns (Response);
}

message Response {
  string status = 1;
}
